sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/Spark33XShims.scala

```diff
@@ -312,19 +312,116 @@ trait Spark33XShims extends Spark321PlusShims with Spark320PlusNonDBShims {
         (a, conf, p, r) => new BinaryAstExprMeta[LessThanOrEqual](a, conf, p, r) {
           override def convertToGpu(lhs: Expression, rhs: Expression): GpuExpression =
             GpuLessThanOrEqual(lhs, rhs)
+        }),
+        GpuOverrides.expr[Abs](
+          "Absolute value",
+          ExprChecks.unaryProjectAndAstInputMatchesOutput(
+            TypeSig.implicitCastsAstTypes, TypeSig.gpuNumeric + TypeSig.ansiIntervals,
```

Comment:
AST supports durations, so there should be updated checks to allow this on the AST and tests added to verify the behavior.  Can be a followup if desired.

Reply:
ok

Reply:
The following 4 operators are ANSI operators, which means they will throw exception when overflow occuring.
Abs: abs(interval)
UnaryMinus: -interval
Add: interval + interval
Subtract: interval - interval

For the `UnaryPositive`  it can be put in AST.